components/common/Header.js

```diff
@@ -1,91 +1,99 @@
+            src="/static/images/logo-turq-transparent.png"
+            alt="Wir.by logo"
+          />
+        </a>
+      </Link>
+      <div id="navburger" className="button navbar-burger">
+        <span />
+        <span />
+        <span />
+      </div>
     </div>
     <div id="navmenu" className="navbar-menu">
       <div className="navbar-start">
-        {ROUTES.map(({ NavLink = Link, id, href, value }) => (
-          <NavLink key={id} href={href}>
+        {NAVBAR_ROUTES.map(({ name, label, pattern = name, params, isActive = checkActive }) => (
+          <Link key={name} route={name} params={params}>
             <a className="navbar-item">
-              <span className={classNames('rubric', { 'is-active': router.asPath === href })}>
-                {value}
+              <span className={classNames('rubric', { 'is-active': isActive(asPath, pattern) })}>
+                {label.toUpperCase()}
               </span>
             </a>
-          </NavLink>
+          </Link>
         ))}
       </div>
 
       <div className="navbar-end">
         <div className="navbar-item">
-          <div className="user">
-            <span>{user && user.email}</span>
-            <div>
-              <Clickable
-                tag="a"
-                className="logout"
-                onClick={() => (user ? signOut() : router.push(login))}
-              >
-                {user ? text.signOutTitle : text.sigInTitle}
-              </Clickable>
-            </div>
-            <div>
-              <div className="dropdown is-right is-hoverable">
-                <div className="dropdown-trigger">
-                  <span className="current-lang">беларуская</span>
+          <LocaleContext.Consumer>
+            {lang => (
+              <div className="user">
+                {/* TODO: to replace user.email with user.displayName once supported by API */}
+                <span>{user && user.email}</span>
+                <div className="auth-section">
+                  <Clickable
+                    tag="a"
+                    className="auth-button"
+                    onClick={() =>
+                      user ? signOut() : Router.pushRoute(ROUTES_NAMES.login, { lang })
+                    }
+                  >
+                    {user ? text.signOutTitle : text.sigInTitle}
+                  </Clickable>
+                </div>
+                <div className="lang-section">
+                  <div className="dropdown is-right is-hoverable">
+                    <div className="dropdown-trigger">
+                      <span className="current-lang">{LOCALES[lang]}</span>
+                    </div>
+                    <div className="dropdown-menu" role="menu">
+                      <div className="dropdown-content">
+                        {LANGS.filter(({ id }) => id !== lang).map(({ id, label }) => (
+                          <Link route={getLocaleSwitchUrl(asPath, id)} params={{ lang: id }}>
```

Comment:
There is an error in console `` Each child in an array or iterator should have a unique "key" prop.``